### DISCUSSION TITLE
After upgrading to SDK 49, the XMLHttpRequest progress event is no longer triggered.

### DISCUSSION THREAD

**Discussion by Makeit-Tipi** [expo/expo#27448] [Archived: Questions]
I recently decided to upgrade my project from Expo SDK 48 to SDK 50. However, since the upgrade, the progress event on the upload is no longer being called.

When I revert back to SDK 48, I can see that the event is still being called. However, the event disappears as soon as we upgrade to SDK 49.

Does anyone have an idea of what has been changed?

```
const file = data.file;

    const xhr = new XMLHttpRequest();
    // no need to reject
    const xhrPromise = new Promise((resolve) => {
      xhr.onreadystatechange = function () {
        if (xhr?.readyState === 4) {
          if (xhr?.status === 200) {
            // Successfully uploaded the file.
            if (config.local) {
              console.log('successfully uploaded presigned url', file);
            }
            resolve(xhr);
          } else {
            // The file could not be uploaded.
            if (config.local) {
              console.log('failed to upload presigned url', file);
            }
            resolve(xhr);
          }
        }
      };
      // Here we impose the method
      xhr.open('PUT', url);

      xhr.upload.addEventListener('progress', function (event) {
        if (options.progress && typeof options.progress === 'function' && event.lengthComputable) {
          options.progress(event);
        }
      });
      // If no X-Amz-ACL header is sent, the one from the upload url will be used
      if (!options.private) {
        console.warn('Header sending file to public bucket');
        xhr.setRequestHeader('X-Amz-ACL', 'public-read');
      }

      if (data?.mime) {
        xhr.setRequestHeader('Content-Type', data?.mime);
      }

      xhr.send(file);
    });
    return xhrPromise.then((resp) => resp);
```
        
Here are the package updated when upgrading SDK 48 => 49 
<img width="1191" alt="Screenshot 2024-03-05 at 16 34 49" src="https://github.com/expo/expo/assets/88028094/724bcf4f-dfa2-467a-9a07-ccfb1f7f9db1">

**Comment by ChristopherGabba** [comment]
+1 on this issue. We are seeing this on our Amplify uploadData function as well, and have traced the issue. It works on Expo 48 but not on 49 and 50.

Another open issue with expo for similar topic:
https://github.com/expo/expo/issues/28269 

And noticed this issue with another package that I'm also seeing:
https://github.com/numandev1/react-native-compressor/issues/276

**Comment by jhau** [comment]
Any workaround for this issue?

**Comment by ChristopherGabba** [comment]
No I haven't found one. Still no xml progress updates.

**Comment by ChristopherGabba** [comment]
Any update on this issue? This still continues to be an annoyance in my app on uploads where there is no progress.

**Comment by markrickert** [comment]
I made an actual reproduction example that rate-limits a 4+mb file upload to a local server so that it takes much longer than usual - expo 48 fires the progress events but expo 49+ does not. I've got a comment on this issue and a linked minimal reproducible example here: https://github.com/expo/expo/issues/28269#issuecomment-2845041078
